eox_nelp/signals/receivers.py

```diff
@@ -283,3 +288,49 @@ def emit_subsection_attempt_event(usage_id, user_id, *args, **kwargs):  # pylint
         usage_id=usage_id,
         user_id=user_id
     )
+
+
+def mt_course_completion_handler(instance, **kwargs):  # pylint: disable=unused-argument
+    """This receiver is connected to the post_save BlockCompletion signal and this executes
+    the course_completion_mt_updater task, that basically updates the training stage based
+    on the completion logic.
+
+    Arguments:
+        instance<Blockcompletion>: Instance of BlockCompletion model.
+    """
+    course_completion_mt_updater.delay(
```

Comment:
It is possible to avoid this sent only for mt with the settings. I mean this kind of  line
https://github.com/eduNEXT/eox-nelp/blob/6671b660204aee841c1a95899ea86d26136473a0/eox_nelp/signals/tasks.py#L250-L251
but here.

Because I think that is better to create only tasks related to mt in the worker, and not all tenants that have completion events or  COURSE_GRADE_NOW_PASSED or  COURSE_GRADE_NOW_FAILED

Reply:
I added that in that way, since I didn't want to repeat code, however I prefer having repeated code instead of   unnecessary tasks as you pointed out

Reply:
You could create a function like `is_mt_completion_activated` xD or repeat the six lines if it doesn't deserve

Reply:
that means that I have to repeat the method's call,  that makes no difference